payjoin-directory/src/config.rs

```diff
@@ -0,0 +1,46 @@
+use std::path::PathBuf;
+use std::time::Duration;
+
+use anyhow::Result;
+use config::builder::DefaultState;
+use config::{ConfigError, File, FileFormat};
+use serde::Deserialize;
+
+type Builder = config::builder::ConfigBuilder<DefaultState>;
+
+use crate::cli::Cli;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Config {
+    pub listen_addr: String, // TODO tokio_listener::ListenerAddressLFlag
+    pub timeout: Duration,
+    pub db_host: String,
+    pub ohttp_keys: PathBuf, // TODO OhttpConfig struct with rotation params, etc
+}
+
+impl Config {
+    pub fn new(cli: &Cli) -> Result<Self, ConfigError> {
+        let mut config = config::Config::builder();
+        config = add_defaults(config, cli)?;
+
+        // what directory should this reside in? require explicit --config-file? ~/.config? /etc?
+        config = config.add_source(File::new("config.toml", FileFormat::Toml).required(false));
```

Comment:
I think we should follow the same pattern as payjoin-cli for configuration file handling:

first, check for config file in  xdg-compliant default directory
fall back to current working directory if no XDG directory is found
allow --config-file flag to override both defaults when specified

This would provide a consistent user experience across the codebase and follow established conventions.